src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -536,6 +537,94 @@ def load_sub_model(
     return loaded_sub_model
 
 
+def _load_empty_model(
+    library_name: str,
+    class_name: str,
+    importable_classes: List[Any],
+    pipelines: Any,
+    is_pipeline_module: bool,
+    pipeline_class: Any,
+    name: str,
+    torch_dtype: Union[str, torch.dtype],
+    cached_folder: Union[str, os.PathLike],
+    **kwargs,
+):
+    # retrieve class candidates
+    class_obj, class_candidates = get_class_obj_and_candidates(
+        library_name,
+        class_name,
+        importable_classes,
+        pipelines,
+        is_pipeline_module,
+        component_name=name,
+        cache_dir=cached_folder,
+    )
+
+    if is_transformers_available():
+        transformers_version = version.parse(version.parse(transformers.__version__).base_version)
+    else:
+        transformers_version = "N/A"
+
+    # Determine library.
+    is_transformers_model = (
+        is_transformers_available()
+        and issubclass(class_obj, PreTrainedModel)
+        and transformers_version >= version.parse("4.20.0")
+    )
```

Comment:
I think that we can possibly offload this work inside of accelerate ! I know that @muellerzr  worked on something similar to measure the size of transformers and timm models (see related [doc](https://huggingface.co/docs/accelerate/main/en/usage_guides/model_size_estimator) and [function](https://github.com/huggingface/accelerate/blob/70cdf5fe526f5651347da8b9ce6e753c6d028b4d/src/accelerate/commands/estimate.py#L64) ). Maybe we can add diffusers models with what we have here !

Reply:
Oh I think it doesn't support `diffusers` yet.

Reply:
Yes, i'm thinking that it could be better if we add the support for diffusers model there. WDYT ? Is there any big blocker compared to doing it here ?

Reply:
No, absolutely not. But I will let the `accelerate` folks take care of that if that's okay? But happy to contribute to `accelerate` but I am a little limited on bandwidth. LMK.

Reply:
Sounds good. I'll have a look then on how to integrate what you did here.